### ISSUE TITLE
valid-jsdoc cannot handle ArrayType

### ISSUE THREAD

**Issue by BYK** [eslint/eslint#5746]
**What version of ESLint are you using?**

2.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default (Espree)

**Please show your full configuration:**

Only pasting the relevant part:
```yaml
valid-jsdoc:
        - 2
        - requireReturn: false
          requireReturnDescription: false
          prefer:
              return: returns
          preferType:
              Any: '*'
              String: string
              Number: number
              Boolean: boolean
              Function: function
              RegExp: regexp
              date: Date
              array: Array
              object: Object
              promise: Promise
```

**What did you do? Please include the actual source code causing the issue.**

Tried to validate the following snippet:

```js
/**
  * Some explanation
  * @param {Array.<[string, number]>} data - Array of [string, number] arrays.
  * @param {string} interval - A valid interval string
  * @returns {Object} - An object with date intervals as the key and
  *                     the corresponding value summed over interval
  */
  formatDataByInterval: function (data, interval) {
    some.serious.code();
}
```

**What did you expect to happen?**

The code to pass validation without issues.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint crashed with the following stack trace:
```
Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at getCurrentExpectedTypes (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\rules\valid-jsdoc.js:111:42)
    at C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\rules\valid-jsdoc.js:153:35
    at Array.forEach (native)
    at validateTagType (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\rules\valid-jsdoc.js:150:18)
    at C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\rules\valid-jsdoc.js:272:21
    at Array.forEach (native)
    at EventEmitter.checkJSDoc (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\rules\valid-jsdoc.js:208:24)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.leaveNode (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\util\node-event-generator.js:51:22)
    at CodePathAnalyzer.leaveNode (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:628:23)
    at CommentEventGenerator.leaveNode (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\util\comment-event-generator.js:111:23)
    at Controller.traverser.traverse.leave (C:\Users\BYK\Documents\Projects\Disqus\next\node_modules\eslint\lib\eslint.js:872:36)
```

**Comment by BYK** [comment]
Working on this.
